### ISSUE TITLE
Revise allowed roles for label element

### ISSUE THREAD

**Issue by scottaohara** [w3c/html-aria#492]
the rules around what roles are allowed on label should better refelct when adjusting the role will cause an accessibility issue. 

there are really 3 differnet use cases:

1. label is associated with a form element via its for attribute
2. label is associated with a form element by being its ancestor
3. label is not associated with a form element

For 1 & 2, there can be accName breakages if the role is changed - and for 2 specifically there can be potential invalid nesting of role issues we would want to avoid

For 3.... someone decided to use `label` instead of `span`.  For most cases, I'd submit there's not a whole lot that can go wrong here if someone were to modify the role so the element could be exposed as / behave as something else.
